File: src/main/java/co/bitshifted/xapps/backstage/BackstageConstants.java
/*
 * Copyright (c) 2020. Bitshift (http://bitshifted.co)
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

package co.bitshifted.xapps.backstage;

import java.time.ZoneId;

/**
 * Constants used across application.
 *
 * @author Vladimir Djurovic
 */
public final class BackstageConstants {

	private BackstageConstants() {
		// private constructor to prevent instantiation
	}

	public static final String DEPLOYMENT_CONTENT_TYPE = "application/zip";

	/**
	 * HTTP header containing URL of deployment status.
	 */
	public static final String DEPLOYMENT_STATUS_HEADER = "X-Deployment-Status";

	/**
	 * UTC timezone ID.
	 */
	public static final ZoneId UTC_ZONE_ID = ZoneId.of("UTC");

	/**
	 * UTC time zone ID as a string.
	 */
	public static final String UTC_ZONE_ID_STRING = "UTC";

	/**
	 * File name of update archive for JRE modules
	 */
	public static final String MODULES_UPDATE_FILE_NAME = "modules.zip";

	public static final String MODULES_UPDATE_ZSYNC_FILE_NAME = MODULES_UPDATE_FILE_NAME + ".zsync";

	/**
	 * File name of update archive for application content.
	 */
	public static final String CONTENT_UPDATE_FILE_NAME = "contents.zip";

	public static final String CONTENT_UPDATE_ZSYNC_FILE_NAME = CONTENT_UPDATE_FILE_NAME + ".zsync";

	/**
	 * Name of update information file.
	 */
	public static final String UPDATE_INFO_FILE_NAME = "update-info.xml";

	/**
	 * MIME type of Zsync control file
	 */
	public static final String ZSYNC_MIME_TYPE = "application/x-zsync ";

	public static final String IGNITE_CONFIG_FILE_NAME = "ignite-config.xml";

	public static final String LAUNCHER_CONFIG_FILE_NAME = "application.xml";

	public static final String LAUNCHER_FILE_NAME_MAC = "launchcode-mac-x64";
	public static final String LAUNCHER_FILE_NAME_WIN_64 = "launchcode-windows-x64";

	/**
	 * Comment to set on executable files when they are added to .zip archives.
	 */
	public static final String ZIP_ENTRY_EXEC_COMMENT = "exec:true";

	public static final String DEPLOY_PKG_MODULES_DIR_NAME = "modules";
	public static final String DEPLOY_PKG_CLASSPATH_DIR_NAME = "classpath";
	public static final String JDK_JMODS_DIR_NAME = "jmods";
}

File: src/main/kotlin/co/bitshifted/backstage/BackstageConstants.kt
/*
 *
 *  * Copyright (c) 2022  Bitshift D.O.O (http://bitshifted.co)
 *  *
 *  * This Source Code Form is subject to the terms of the Mozilla Public
 *  * License, v. 2.0. If a copy of the MPL was not distributed with this
 *  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 *
 */

package co.bitshifted.backstage

import java.time.ZoneId

object BackstageConstants {

    /**
     * UTC timezone ID.
     */
     val UTC_TIME_ZONE = ZoneId.of("UTC")

    val JDK_MODULES_PREFIXES = listOf("java.", "jdk.")

    /**
     * HTTP header containing URL of deployment status.
     */
    const val DEPLOYMENT_STATUS_HEADER = "X-Deployment-Status"

    const val THREAD_POOL_KEEP_ALIVE_MS = 500L

    const val DEPLOYMENT_DEPENDENCIES_DIR = "dependencies"
    const val DEPLOYMENT_RESOURCES_DIR = "resources"
    const val DEPLOYMENT_OUTPUT_DIR = "output"
    const val DEPLOYMENT_CONFIG_FILE = "deployment.json"
    const val OUTPUT_CLASSPATH_DIR = "cp"
    const val OUTPUT_MODULES_DIR = "modules"
    const val OUTPUT_JRE_DIR = "jre"
    const val OUTPUT_LAUNCHER_DIR = "launchcode"
    const val OUTPUT_LAUNCHER_DIST_DIR = "dist"
    const val JAR_EXTENSION = ".jar"
    const val JDK_JMODS_DIR_NAME = "jmods"
    const val LATEST_JAVA_DIR_LINK = "latest"
    const val LAUNCHER_NAME_LINUX = "launchcode-linux"
    const val LAUNCHER_NAME_MAC = "launchcode-mac"
    const val LAUNCHER_NAME_WINDOWS = "launchcode-windows.exe"
}